package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -23,9 +24,13 @@ def get_catalog(self) -> DataCatalog:
     def set_catalog(self, value: DataCatalog):
         self._catalog = value
 
-    @staticmethod
-    def strip_encoding(dataset_name: str) -> str:
-        return dataset_name.split("@")[0]
+    def _validate_layers_for_transcoding(self, dataset_name, layer):
+        existing_layer = self._layers_mapping.get(dataset_name)
+        if existing_layer is not None and existing_layer != layer:
```

Comment:
```suggestion
        if existing_layer != layer:
```
If we use 
`existing_layer = self._layers_mapping.setdefault(dataset_name, layer)`
in the line above, we can shorten the if condition, and drop the `self._layers_mapping[dataset_name] = layer` after the if conditions on lines 76 and 95

Reply:
as discussed, we will keep the function as is so that's it does what it says i.e. purely validates if the layer for transcoding datasets and raises an error if two datasets have different layers.